.build/docker/run-tests.sh

```diff
@@ -0,0 +1,259 @@
+        [[ ${mem} -gt $((5 * 1024 * 1024 * 1024 * ${jenkins_executors})) ]] || { echo >&2 "tests require minimum docker memory 6g (per jenkins executor (${jenkins_executors})), found ${mem}"; exit 1; }
+        docker_runs=1
+    ;;
+    "dtest" | "dtest-novnode" | "dtest-offheap" | "dtest-large" | "dtest-large-novnode" | "dtest-upgrade" )
+        [ -f "${cassandra_dtest_dir}/dtest.py" ] || { echo >&2 "${cassandra_dtest_dir}/dtest.py must exist"; exit 1; }
+        [[ ${mem} -gt $((15 * 1024 * 1024 * 1024 * ${jenkins_executors})) ]] || { echo >&2 "dtests require minimum docker memory 16g (per jenkins executor (${jenkins_executors})), found ${mem}"; exit 1; }
+        docker_runs=1
+        test_script="dtest-python.sh"
+        docker_mounts="${docker_mounts} -v ${cassandra_dtest_dir}:/home/cassandra/cassandra-dtest"
+        # check that exists ${cassandra_dtest_dir}
+        [ -f "${cassandra_dtest_dir}/dtest.py" ] || { echo >&2 "${cassandra_dtest_dir}/dtest.py not found. please specify 'cassandra_dtest_dir' to point to the local cassandra-dtest source"; exit 1; }
+    ;;
+    "test"| "test-cdc" | "test-compression" | "jvm-dtest" | "jvm-dtest-upgrade")
+        [[ ${mem} -gt $((5 * 1024 * 1024 * 1024 * ${jenkins_executors})) ]] || { echo >&2 "tests require minimum docker memory 6g (per jenkins executor (${jenkins_executors})), found ${mem}"; exit 1; }
+        max_docker_runs_by_cores=$( echo "sqrt( ${cores} / ${jenkins_executors} )" | bc )
+        max_docker_runs_by_mem=$(( ${mem} / ( 5 * 1024 * 1024 * 1024 * ${jenkins_executors} ) ))
+        docker_runs=$(( ${max_docker_runs_by_cores} < ${max_docker_runs_by_mem} ? ${max_docker_runs_by_cores} : ${max_docker_runs_by_mem} ))
+        docker_runs=$(( ${docker_runs} < 1 ? 1 : ${docker_runs} ))
+    ;;
+    *)
+    echo "unrecognized \"${target}\""
+    exit 1
+    ;;
+esac
+
+docker_runs=1 # tmp FIXME
+
+# Break up the requested split chunk into a number of concurrent docker runs, as calculated above
+# This will typically be between one to four splits. Five splits would require >25 cores and >25GB ram
+inner_splits=$(( $(echo $split_chunk | cut -d"/" -f2 ) * ${docker_runs} ))
+inner_split_first=$(( ( $(echo $split_chunk | cut -d"/" -f1 ) * ${docker_runs} ) - ( ${docker_runs} - 1 ) ))
+docker_cpus=$(echo "scale=2; ${cores} / ( ${jenkins_executors} * ${docker_runs} )" | bc)
+
+# hack: long-test does not handle limited CPUs
+if [ "${target}" == "long-test" ] ; then
+    docker_flags="-m 5g --memory-swap 5g"
+elif [[ "${target}" =~ dtest* ]] ; then
+    docker_flags="--cpus=${docker_cpus} -m 15g --memory-swap 15g"
+else
+    docker_flags="--cpus=${docker_cpus} -m 5g --memory-swap 5g"
+fi
+docker_flags="${docker_flags} --env-file build/env.list -d --rm"
+
+# make sure build_dir is good
+mkdir -p ${cassandra_dir}/build/tmp || true
+mkdir -p ${cassandra_dir}/build/test/logs || true
+chmod -R ag+rwx ${cassandra_dir}/build
+
+# FIXME python
```

Comment:
What's the word here? Why FIXME?

Reply:
the test scripts are still wip.

here, testing different python versions isn't implemented.

Reply:
Gotcha. Any chance we could augment the comment here in the interim with something about what FIXME looks like in the future?